meilisearch-http/src/routes/setting.rs

```diff
@@ -39,12 +39,31 @@ pub async fn get(ctx: Context<Data>) -> SResult<Response> {
 
     let settings = match index.main.customs(&reader).unwrap() {
         Some(bytes) => bincode::deserialize(bytes).unwrap(),
-        None => SettingBody::default(),
+        None => Setting::default(),
     };
 
     Ok(tide::response::json(settings))
 }
 
+#[derive(Default, Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase", deny_unknown_fields)]
+pub struct SettingBody {
+    #[serde(default, deserialize_with = "deserialize_some")]
+    pub ranking_order: Option<Option<RankingOrder>>,
+    #[serde(default, deserialize_with = "deserialize_some")]
+    pub distinct_field: Option<Option<DistinctField>>,
+    #[serde(default, deserialize_with = "deserialize_some")]
+    pub ranking_rules: Option<Option<RankingRules>>,
+}
+
+// Any value that is present is considered Some value, including null.
+fn deserialize_some<'de, T, D>(deserializer: D) -> Result<Option<T>, D::Error>
+    where T: Deserialize<'de>,
+          D: Deserializer<'de>
+{
+    Deserialize::deserialize(deserializer).map(Some)
+}
+
```

Comment:
What is that new struct ?

Reply:
Does it work? Could you provide a screenshot?

Reply:
Did you looked at `#[serde(flatten)]`?

It is probably what you want.
https://github.com/serde-rs/serde/issues/1255

Reply:
Flatten is not what I want. I need to know if the user has omitted the field or if he has set it to null. I used the example from here https://github.com/serde-rs/serde/issues/984